ncm-systemd/src/main/pan/components/systemd/schema.pan

```diff
@@ -42,6 +42,9 @@ type ${project.artifactId}_unit_virtualization = string with match(SELF,
 #    make this more finegrained, e.g. has to be existing unit; or check types
 type ${project.artifactId}_valid_unit = string;
 
+# modes are in octal
+type ${project.artifactId}_systemd_mode = string with match(SELF, '^0?1?[0-7][0-7][0-7]');
```

Comment:
Why not just `... = string with to_long(SELF, 8) >= 0 && to_long(SELF, 8) <= 07777`? That's what `systemd`'s parser is doing - there are no requirements on the number of digits.

Reply:
even better, thanks for the hint.

I've pushed this to template-library-core#200 and the same change to this PR. As soon as the former is merged, I can switch the type here.